.github/workflows/check_ghcr_push_rights.yml

```diff
@@ -0,0 +1,55 @@
+on:
+  workflow_call:
+    outputs:
+      have_secrets:
+        description: "In possession of ghcr.io tokens?"
+        value: ${{ jobs.check_push_rights.outputs.have_secrets }}
+
+# No concurrency group or cancel-in-progress here as this workflow is called from other workflows
+# so if you have a concurrency definition here it will be considered as a deadlock and job dismissed
+# The job below is also very fast, so no real need for cancel-in-progress
+
+jobs:
+  check_push_rights:
+    runs-on: ubuntu-latest
+
+    outputs:
+      have_secrets: ${{ steps.check-secrets.outputs.have_secrets }}
+
+    steps:
+    #Check we have access to secrets for pushing to GHCR. Forks do not
+    - name: check GITHUB_TOKEN allows GHCR push access
+      id: check-secrets
+      run: |
+          if [[ ("${{ github.event_name }}" == "push") && \
+                (("${{ github.repository }}" == "eclipse/kuksa.val") || \
+                 ("${{ github.repository }}" == "eclipse/kuksa.val.feeders") || \
+                 ("${{ github.repository_owner }}" == "eclipse-kuksa")) ]] ; then
+            echo "We are pushing to kuksa.val upstream, so we should have rights"
+            echo "have_secrets=true" >> $GITHUB_OUTPUT
+            exit 0
+            # if it is a pull_request and my_repo is kuksa.val I can push to GHCR,
+            # (note that some/all workflows int his repo might still opt to no push PR builds to GHCR)
+          fi
+          if [[ ("${{ github.event_name }}" == "pull_request") && \
+                (("${{ github.event.pull_request.head.repo.full_name }}" == "eclipse/kuksa.val") ||\
+                 ("${{ github.event.pull_request.head.repo.full_name }}" == "eclipse/kuksa.val.feeders") || \
+                 ("${{ github.event.pull_request.head.repo.owner.login }}" == "eclipse-kuksa")) ]] ; then
+              echo "We are an internal pull request , so we should have rights"
```

Comment:
I do not know if any build tries to push to ghcr for pull requests (if we get true here). We could argue that it is not good practice - we can as well use ttl.sh for that purpose. In some (all?) builds we already have an extra check like:

`if: ${{ needs.checkrights.outputs.have_secrets == 'true' && github.event_name != 'pull_request' }}`


But if we change - should we then possibly rename it so it rather returns `push_ghcr=true` and call it `check_ghcr_push`. That would simplify usage and would make it easier for us to change behavior for all kuksa repos using this workflow.

Reply:
I agree. So we do not push any PRs to GHCR anymore, and it is also not really recommended I would say, as there are some quotas in play. So this action really decides "Should we push to GHCR?" depending on whether we have the rights and whether we _want_ to

Reply:
I can refactor it and try it out